tests/sentry/hybridcloud/test_caching.py

```diff
@@ -158,3 +161,118 @@ def cache_death_event() -> Generator[None, None, None]:
     random = Random(84716393)
     for i in range(10000):
         next(random.choice([reader1, reader2, writer1, cache_death]))
+
+
+@django_db_all(transaction=True)
+def test_caching_many() -> None:
+    cache.clear()
+
+    @back_with_silo_cache_many(base_key="get_users", silo_mode=SiloMode.REGION, t=RpcUser)
+    def get_users(user_ids: list[int]) -> list[RpcUser]:
+        return user_service.get_many(filter=dict(user_ids=user_ids))
+
+    users = [Factories.create_user() for _ in range(3)]
+    user_ids = [u.id for u in users]
+
+    wrapped_result = get_users(user_ids)
+    direct_result = get_users.cb(user_ids)
+    assert len(wrapped_result) == len(direct_result)
+    for wrapped, direct in zip(wrapped_result, direct_result):
+        assert wrapped == direct
+
+    with assume_test_silo_mode(SiloMode.CONTROL):
+        for user in users:
+            user.update(username=user.username + "moo")
+
+    # Does not include updates made to db
+    after_update_wrapped = get_users(user_ids)
+    for u in after_update_wrapped:
+        assert not u.username.endswith("moo")
+        # Clear cache simulating outbox logic
+        region_caching_service.clear_key(
+            region_name=get_local_region().name, key=get_users.key_from(u.id)
+        )
```

Comment:
Do we have any integration style tests that assert this behavior e2e with actual outbox/replication logic? I saw the testing in getsentry/sentry#71783 that validates we're calling the cache clear logic, but it's probably valuable to have realistic testing scenarios for this cache behavior if we don't already.

Reply:
I could add that integration testing alongside the first usage of these methods. Without that real usage it is hard to have a end to end scenario.

Reply:
Agreed, that sounds good to me!